packages/cli/src/commands/init/init.ts

```diff
@@ -393,37 +446,13 @@ function checkPackageManagerAvailability(
   return false;
 }
 
-function getNpmRegistryUrl(): string {
-  try {
-    return execSync('npm config get registry').toString().trim();
-  } catch {
-    return 'https://registry.npmjs.org/';
-  }
-}
-
-async function urlExists(url: string): Promise<boolean> {
-  try {
-    // @ts-ignore-line: TS2304
-    const {status} = await fetch(url, {method: 'HEAD'});
-    return (
-      [
-        200, // OK
-        301, // Moved Permanemently
-        302, // Found
-        304, // Not Modified
-        307, // Temporary Redirect
-        308, // Permanent Redirect
-      ].indexOf(status) !== -1
-    );
-  } catch {
-    return false;
-  }
-}
-
-async function createTemplateUri(options: Options): Promise<string> {
+async function createTemplateUri(
+  options: Options,
+  version: string,
+): Promise<string> {
   if (options.platformName) {
     logger.debug('User has specified an out-of-tree platform, using it');
-    return `${options.platformName}@${options.version ?? DEFAULT_VERSION}`;
+    return `${options.platformName}@${version}`;
```

Comment:
Here you're missing the `?? DEFAULT_VERSION` fallback.

Reply:
I've moved this up the call stack:
```
initialize() -> createProject({ version }) -> createTemplateUri(options, version)
```
In `initialize` I set the `version` that's passed down to:
```typescript
  const version = await npmResolveConcreteVersion(
    'react-native',
    options.version ?? DEFAULT_VERSION,
  );
```